src/main/java/org/embulk/output/elasticsearch/ElasticsearchOutputPluginDelegate.java

```diff
@@ -0,0 +1,226 @@
+package org.embulk.output.elasticsearch;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+import com.google.common.base.Throwables;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.embulk.base.restclient.RestClientOutputPluginDelegate;
+import org.embulk.base.restclient.RestClientOutputTaskBase;
+import org.embulk.base.restclient.jackson.JacksonServiceRequestMapper;
+import org.embulk.base.restclient.jackson.JacksonTaskReportRecordBuffer;
+import org.embulk.base.restclient.jackson.JacksonTopLevelValueLocator;
+import org.embulk.base.restclient.jackson.scope.JacksonAllInObjectScope;
+import org.embulk.base.restclient.jackson.scope.JacksonDirectIntegerScope;
+import org.embulk.base.restclient.jackson.scope.JacksonDirectStringScope;
+import org.embulk.base.restclient.record.RecordBuffer;
+import org.embulk.config.Config;
+import org.embulk.config.ConfigDefault;
+import org.embulk.config.ConfigDiff;
+import org.embulk.config.ConfigException;
+import org.embulk.config.TaskReport;
+import org.embulk.spi.Exec;
+import org.embulk.spi.Schema;
+import org.embulk.util.retryhelper.jetty92.Jetty92ClientCreator;
+import org.embulk.util.retryhelper.jetty92.Jetty92RetryHelper;
+import org.slf4j.Logger;
+
+import java.util.List;
+import java.util.Locale;
+
+public class ElasticsearchOutputPluginDelegate
+        implements RestClientOutputPluginDelegate<ElasticsearchOutputPluginDelegate.PluginTask>
+{
+    private final Logger log;
+    private final ElasticsearchUtils utils;
+
+    public ElasticsearchOutputPluginDelegate()
+    {
+        this.log = Exec.getLogger(getClass());
+        this.utils = new ElasticsearchUtils();
+    }
+
+    public interface NodeAddressTask
+            extends RestClientOutputTaskBase
```

Comment:
Some `base-restclient` methods and classes require this through a generics parameter.